content/automate/ManageIQ/Transformation/TransformationHosts/Common.class/__methods__/utils.rb

```diff
@@ -0,0 +1,66 @@
+#
+# 
+#
+
+module Transformation
+  module TransformationHost
+    module Common
+      class Utils
+        def initialize(handle = $evm)
+          @debug = true
+          @handle = handle
+        end
+
+        def get_transformation_hosts(ems)
+          ems.hosts.select { |host| host.tagged_with('mf_transformation_host', 'true') }
+        end
+        
+        def get_runners_count_by_host(host)
+          return @handle.vmdb(:vm).all.select { |v| v.custom_get('Transformation Host') == host.name }.size
+        end
+        
+        def get_runners_count_by_ems(ems)
+          total_runners = 0
+          get_transformation_hosts.each { |host| total_runners += get_runners_count_by_host(host) }
+          return total_runners
+        end
+
+        def set_transformation_host(vm, ems, factory_config)
+          # Determine the least utilized conversion host
+          min_runners, transformation_host = nil, nil
+          
+          get_transformation_hosts.each do |host|
+            runners_count = get_runners_count_by_host(host)
+            max_runners = host.custom_get('Max Transformation Runners')
+            max_runners = factory_config['transformation_host_max_runners'] if max_runners.blank?
+            next if runners_count == max_runners
+            ( transformation_host = host ; break ) if runners_count == 0
+            if transformation_host.nil? || runners_count < min_runners
```

Comment:
@fdupont-redhat How is the min_runners being set, it starts out as nil and only gets set inside of the if block to the runner count. The || logic seems odd.